src/components/Button/Button.jsx

```diff
@@ -0,0 +1,138 @@
+import { calculateWPM, handleWordReset } from "../WPM/Wpm";
+import PropTypes from 'prop-types';
+import RiseLoader from "react-spinners/RiseLoader";
+import style from "./Button.module.css";
+import KeyboardLayout from '../Keyboard/KeyboardLayout';
+
+const SPECIAL_KEYS = {
+  SPACE: ' ',
+  CONTROL: 'Control',
+  BACKSPACE: 'Backspace'
+};
+
+function Button({ word, wpm, setword, loading }) {
+  let typedText = "";
+  let startTime;
+
+  const handleKeyPress = (event) => {
+    if (!startTime) {
+      startTime = performance.now();
+    }
+
+    const targetValueInput = document.querySelector("#typeWords");
+    if (!targetValueInput) return; // Early return if target input doesn't exist
+
+    const keyIdentify = event.key;
+    const sanitizedKey = getSanitizedKey(keyIdentify);
+    
+    updateKeyboardVisuals(sanitizedKey);
+    handleTyping(keyIdentify, targetValueInput, event);
+  };
+
+  const getSanitizedKey = (key) => {
+    let sanitized = key.replace(/([.*+?^=${}()`|[\]/\\-])/g, '\\$1');
+    if (key === SPECIAL_KEYS.SPACE) return 'Space';
+    if (key === SPECIAL_KEYS.CONTROL) return 'Ctrl';
+    return sanitized;
+  };
+
+  const updateKeyboardVisuals = (key) => {
+    const isSpecialKey = ["Ctrl", "Shift", "Alt"].includes(key);
+    const target = document.querySelector(`#${key.toLowerCase()}`);
+    
+    if (isSpecialKey) {
+      const leftKey = document.querySelector(`#L${key.toLowerCase()}`);
```

Comment:
The left key is not captured properly.
```suggestion
      const leftKey = document.querySelector(`#l${key.toLowerCase()}`);
```